type-definitions/immutable.js.flow

```diff
@@ -390,7 +400,7 @@ declare class SetSeq<T> extends Seq<T,T> mixins SetIterable<T> {
 }
 
 declare class List<T> extends IndexedCollection<T> {
-  static <T>(iterable?: ESIterable<T>): List<T>;
+  static (iterable?: ESIterable<T>): List<T>;
```

Comment:
We need it when we bind another dynamic type.
```javascript
declare class Foo<T> {
 static (param: T): Foo<T>; //not needed.
 static <T_>(param: T, cb: (p: T, extra: T_) => T_): Foo<T> // needed - but only T_ not T.
}
```

The TypeParam of the class we define the static on is implicitly available.